### PR TITLE
FS2-2550: Fix errors when creating small multiples

### PR DESCRIPTION
**Pull request by chschan**

### CODE REVIEW

File: R/combinedscatterplot.R [hunk 1]
```diff
@@ -397,8 +416,8 @@ CombinedScatter <- function(x = NULL,
         fit.y = fit$fit.y,
         fit.group = fit$fit.group,
         fit.panel = fit$fit.panel,
-        fit.lower.bound = fit$fit.lower.bound,
-        fit.upper.bound = fit$fit.upper.bound,
+        fit.lower.bound = if (fit.CI.show) fit$fit.lower.bound else NULL,
```
Comment (JustinCCYap) [inline R/combinedscatterplot.R:419]: I encountered strange issues with using inline `if else` when passing parameters, but I assume it works for you?

File: R/combinedscatterplot.R [hunk 2]
```diff
@@ -19,6 +24,8 @@ CombinedScatter <- function(x = NULL,
                             scatter.colors.name = NULL,
                             scatter.colors.column = 4,
                             scatter.colors.as.categorical = TRUE,
+                            scatter.groups = NULL,
```
Comment (JustinCCYap) [inline R/combinedscatterplot.R:27]: I assume changes will need to be made to the Plugins code to use these new parameters?

Comment (JustinCCYap) [reply R/combinedscatterplot.R:27]: And that this is better than using the previous way of specifying groups which had issues?

File: R/combinedscatterplot.R [hunk 3]
```diff
@@ -523,9 +550,9 @@ processScatterSizes <- function(scatter.sizes, n) {
     sz.tmp
 }
 
-getOpacity <- function(opacity, scatter.sizes) {
+getOpacity <- function(opacity, scatter.sizes, fit.type) {
     if (is.null(opacity)) {
-        if (!is.null(scatter.sizes)) {
+        if (!is.null(scatter.sizes) || fit.type != "None") {
```
Comment (JustinCCYap) [inline R/combinedscatterplot.R:555]: I assume this is what we did before? Why is this needed?

File: R/combinedscatterplot.R (hunk 2 above)
Comment (chschan) [reply R/combinedscatterplot.R:27]: `scatter.groups` is actually the variable that determines how the points are separated into the small multiples. It needs to be this name because that is the same param name as used in the SmallMultiples function. It actually doesn't require a change to Plugins because it already deals with this

The code in `Scatter` sometimes uses a variable called `groups` but this is actually `scatter.colors` which I have renamed to remove confusion

File: R/combinedscatterplot.R (hunk 3 above)
Comment (chschan) [reply R/combinedscatterplot.R:555]: LabeledScatter didn't have this because it never allowed fit lines to be drawn. But Scatter has this code: https://github.com/Displayr/flipStandardCharts/blob/master/R/scatterchart.R#L417

File: R/combinedscatterplot.R (hunk 1 above)
Comment (chschan) [reply R/combinedscatterplot.R:419]: This worked for me

File: tests/testthat/test-combinedscatterplot.R
Comment (JustinCCYap) [inline tests/testthat/test-combinedscatterplot.R:1]: Could you add some tests for line of best fit when there aren't small multiples?

**Review by JustinCCYap — APPROVED** [review]
